src/view.js

```diff
@@ -11,13 +11,180 @@ var isFunction = require("can-util/js/is-function/is-function"),
 	deepAssign = require("can-util/js/deep-assign/deep-assign"),
 	canFrag = require("can-util/dom/frag/frag"),
 	each = require("can-util/js/each/each"),
+	can = require("can-namespace"),
+	Observation = require("can-observation"),
 	// Used for hookup `id`s.
 	hookupId = 1;
 
 // internal utility methods
 // ------------------------
 
+var makeRenderer = function(textRenderer) {
+	if (textRenderer.renderType === "fragment") {
+		return textRenderer;
+	}
+	var renderer = function() {
+		return $view.frag(textRenderer.apply(this, arguments));
+	};
+	renderer.render = function() {
+		return textRenderer.apply(textRenderer, arguments);
+	};
+	return renderer;
+};
+
+// ##### checkText
+// Makes sure there's a template, if not, have `steal` provide a warning.
+var checkText = function (text, url) {
+	if (!text.length) {
+
+		// _removed if not used as a steal module_
+
+		//!steal-remove-start
+		can.dev.log("can/view/view.js: There is no template or an empty template at " + url);
+		//!steal-remove-end
+
+		throw new Error("can.view: No template or empty template:" + url);
+	}
+};
+
+// ##### get
+// get a deferred renderer for provided url
+/**
+ * @hide
+ * @function get
+ * @param {String | Object} obj url string or object with url property
+ * @param {Boolean} async If the ajax request should be asynchronous.
+ * @return {can.Deferred} a `view` renderer deferred.
+ */
+var	getRenderer = function (obj, async) {
+	// If `obj` already is a renderer function just resolve a Deferred with it
+	if(isFunction(obj)) {
+		var def = can.Deferred();
```

Comment:
Could use https://github.com/canjs/canjs/blob/2.3-legacy/util/deferred.js for deferred.